src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -1098,6 +1099,56 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// 1463 - Incanter's Flow
+class spell_mage_incanters_flow : public AuraScript
+{
+    PrepareAuraScript(spell_mage_incanters_flow);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MAGE_INCANTERS_FLOW });
+    }
+
+    void HandlePeriodicTick(AuraEffect const* aurEff)
+    {
+        // Incanter's flow should not cycle out of combat
+        if (!GetTarget()->IsInCombat())
+        {
+            GetTarget()->RemoveAurasDueToSpell(SPELL_MAGE_INCANTERS_FLOW);
+            return;
+        }
+
+        if (Aura* aura = GetTarget()->GetAura(SPELL_MAGE_INCANTERS_FLOW))
+        {
+            uint32 stacks = aura->GetStackAmount();
+            if (aurEff->GetAmount() == 0 || (stacks == 1 && aurEff->GetAmount() == 1)) // 0 = stack up, 1 = remove stack
+            {
+                if (stacks == 5)
+                    const_cast<AuraEffect*>(aurEff)->SetAmount(1);
```

Comment:
Can you store the direction (up/down) as a member of this script instead of having to `const_cast` aurEff?

Reply:
Not what I had in mind